symbolic-base/src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Optimization.hs

```diff
@@ -38,7 +38,7 @@ optimize (ArithmeticCircuit s r w o) = ArithmeticCircuit {
     acRange = optRanges vs r,
     acWitness = (>>= optWitVar vs) <$>  M.filterWithKey (\k _ -> notMember (NewVar k) vs) w,
     acOutput = o <&> \case
-      SysVar sV -> maybe (SysVar sV) ConstVar (M.lookup sV vs)
+      lv@(LinVar k sV b) -> maybe lv (ConstVar . (\t -> k * t + b)) (M.lookup sV vs)
```

Comment:
Actually, being a constant-or-linear expression, `Var` is a monad with `>>=` being a substitution (same as with `WitnessF`). Using this fact, the code here can be simplified to `o <&> (>>= \v -> maybe (pure v) ConstVar (M.lookup v vs))`